extension.json

```diff
@@ -117,57 +117,202 @@
+                    "datastreamIdInputMethod": {
+                      "type": "string",
+                      "enum": ["freeform", "select"]
+                    },
+                    "com_adobe_experience_platform": {
+                      "type": "object",
+                      "properties": {
+                        "datasets": {
+                          "type": "object",
+                          "properties": {
+                            "event": {
+                              "type": "object",
+                              "properties": {
+                                "datasetId": {
+                                  "type": "string"
+                                }
+                              },
+                              "required": ["datasetId"]
+                            }
+                          },
+                          "required": ["event"]
+                        }
+                      },
+                      "required": ["datasets"]
+                    },
+                    "com_adobe_analytics": {
+                      "type": "object",
+                      "properties": {
+                        "reportSuites": {
+                          "type": "array",
+                          "items": {
+                            "type": "string"
                           }
-                        },
-                        "required": ["event"]
-                      }
+                        }
+                      },
+                      "required": ["reportSuites"]
                     },
-                    "required": ["datasets"]
-                  },
-                  "com_adobe_analytics": {
-                    "type": "object",
-                    "properties": {
-                      "reportSuites": {
-                        "type": "array",
-                        "items": {
-                          "type": "string"
+                    "com_adobe_identity": {
+                      "type": "object",
+                      "properties": {
+                        "idSyncContainerId": {
+                          "type": "integer"
                         }
-                      }
+                      },
+                      "required": ["idSyncContainerId"]
                     },
-                    "required": ["reportSuites"]
+                    "com_adobe_target": {
+                      "type": "object",
+                      "properties": {
+                        "propertyToken": {
+                          "type": "string"
+                        }
+                      },
+                      "required": ["propertyToken"]
+                    }
                   },
-                  "com_adobe_identity": {
-                    "type": "object",
-                    "properties": {
-                      "idSyncContainerId": {
-                        "type": "integer"
-                      }
+                  "staging": {
```

Comment:
It would make things a lot easier, but

1. Each of the definitions is technically inside a different json schema and you can't do `$ref` across schemas
2. Even if we could do cross-schema `$ref`, [the schema for `extension.json` doesn't allow for new top-level keys like `$defs`](https://github.com/adobe/reactor-turbine-schemas/blob/master/schemas/extension-package-web.json#L70).

The solution would be to create a script that generates our `extension.json` and have that run as part of our `dev` and `build` scripts. I can do that if you think it's not too much out of scope.